### DISCUSSION TITLE
Multi-Tenant Architecture

### DISCUSSION THREAD

**Discussion by r0x07k** [milvus-io/milvus#36165] [Q&A and General discussion] [answered]
1. What multi-tenancy approach does Milvus recommend for a consumer-grade service where multiple users share the same infrastructure with collection-wise isolation? Is RBAC a suitable solution for this?
2. Could you please provide Java examples for RBAC operations as described in [this documentation](https://milvus.io/docs/rbac.md)?
3. How can Milvus be configured to restrict access to all collections for all users by default, allowing access only to those explicitly granted?
4. Does the `list collections` command return only the collections that the user has access to?

**Comment by xiaofan-luan** [comment, marked as answer]
Milvus's recommendation for multi-tenancy in consumer-grade services:

The choice of multi-tenancy strategy primarily depends on your specific use case, see https://milvus.io/docs/multi_tenancy.md#Multi-tenancy-strategies.  For consumer-grade services where multiple users share the same infrastructure and require collection-level isolation, Milvus recommends using database/collection-level multi-tenancy. This is because:

Your application may need to support several to dozens of use cases.
Each use case (or user) requires independent permission control.
Some use cases may need independent physical resources.

With RBAC, all the access can be achieved. But we also need technical docuemnt for all language SDKs

**Comment by yhmo** [comment]
The Java example for RBAC: https://github.com/milvus-io/milvus-sdk-java/blob/master/examples/main/java/io/milvus/v1/RBACExample.java

**Comment by r0x07k** [comment]
Thank you @xiaofan-luan  and @yhmo!
